site/src/main/java/uk/nhs/digital/common/components/ServiceComponent.java

```diff
@@ -0,0 +1,38 @@
+package uk.nhs.digital.common.components;
+
+import org.hippoecm.hst.content.beans.standard.*;
+import org.hippoecm.hst.core.component.*;
+import org.onehippo.cms7.essentials.components.*;
+import org.slf4j.*;
+
+import java.util.*;
+
+public class ServiceComponent extends EssentialsContentComponent {
+
+    private static Logger log = LoggerFactory.getLogger(ServiceComponent.class);
+
+    @Override
+    public void doBeforeRender(final HstRequest request, final HstResponse response) {
+        super.doBeforeRender(request, response);
+        Object bean = request.getAttribute(REQUEST_ATTR_DOCUMENT);
```

Comment:
Hi @wojtek-oledzki, the EssentialsContentComponent is already doing that (look at the method setContentBeanWith404), what you're suggesting is altering the normal component behavior. The usage of the _instanceof_ is extra check since during the matching phase a folder could be matched.

Reply:
cool! I assume you're planning to handle 404s in future.

Reply:
If you look at the implementation of the setContentBeanWith404, this is already handling the 404. Cheers!